File: yaf-gadomain/src/main/java/ar/edu/ungs/yamiko/ga/domain/impl/BitSetToDoubleRibosome.java
package ar.edu.ungs.yamiko.ga.domain.impl;

import java.io.Serializable;
import java.util.BitSet;

import ar.edu.ungs.yamiko.ga.domain.Ribosome;
import ar.edu.ungs.yamiko.ga.toolkit.BitsStaticHelper;

/**
 * Ribosoma que traduce BitSet a doubles.
 * @author ricardo
 *
 */
public class BitSetToDoubleRibosome implements Ribosome<BitSet>,Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 5472646010255710816L;
	private int floor;
	private int roof;
	private int bitSize;
	
	public BitSetToDoubleRibosome(int floor,int roof,int bitsize) {
		super();
		this.floor = floor;
		this.roof= roof;
		this.bitSize = bitsize;
	}

	public Object translate(BitSet allele) {
		return floor+(roof-floor)*BitsStaticHelper.convertLong(allele)/Math.pow(2, bitSize);
	}
}

File: yaf-gadomain-scala/src/main/scala/ar/edu/ungs/yamiko/ga/domain/impl/BitSetToDoubleRibosome.scala
package ar.edu.ungs.yamiko.ga.domain.impl

import ar.edu.ungs.yamiko.ga.domain.Ribosome
import scala.collection.immutable.BitSet
import ar.edu.ungs.yamiko.ga.toolkit.BitSetHelper

/**
 * Ribosoma que traduce BitSet a enteros.
 * @author ricardo
 *
 */

@SerialVersionUID(1919L)
class BitSetToDoubleRibosome(floor:Int,roof:Int,bitsize:Int) extends Ribosome[BitSet]{
  
	def translate(allele:BitSet):Any = floor+(roof-floor)*BitSetHelper.toLong(allele)/math.pow(2,bitsize)
  
  	
}